### PR TITLE
Authorized teacher data migration

### PR DESCRIPTION
**Pull request by clareconstantine**
Pending comments, this code is done but we don't want to run the data migration until after https://github.com/code-dot-org/code-dot-org/pull/17243 is merged (hence the "do not merge" in the title).

This migration backfills the new UserPermission of Authorized Teacher for previously existing authorized teachers.

https://trello.com/c/wraJEFSl/129-create-authorized-teacher-property-on-user-and-simple-management-page

https://docs.google.com/document/d/15Rk04e0BlGiDZpo82YcQ7qDpRgS5aGPH93IF85hVpR4/edit#

### CODE REVIEW

File: dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class BackfillAuthorizedTeacherUserPermissions < ActiveRecord::Migration[5.0]
+  def up
+    Plc::UserCourseEnrollment.find_each do |enrollment|
+      UserPermission.create(permission: UserPermission::AUTHORIZED_TEACHER, user_id: enrollment.user_id)
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb]: Does it make sense to use [`find_or_create_by`](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by) to convey that there is no need to create a new `UserPermission` if one already exists? Also below.

File: dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    Cohort.find_each do |cohort|
+      cohort.teachers.each do |teacher|
+        UserPermission.create(permission: UserPermission::AUTHORIZED_TEACHER, user_id: teacher.id)
+      end
+    end
+  end
+
+  def down
+    UserPermission.where(permission: UserPermission::AUTHORIZED_TEACHER).destroy_all
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb]: Given that we may have created some `AUTHORIZED_TEACHER` permissions manually between the `up` and the `down`, does it make sense to destroy them all? My expectation would have been to use the default (empty) `down` method.

File: dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb (hunk 1 above)
Comment (clareconstantine) [reply dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb]: That would definitely be clearer. Thanks!

File: dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb (hunk 2 above)
Comment (clareconstantine) [reply dashboard/db/migrate/20170822201210_backfill_authorized_teacher_user_permissions.rb]: I wasn't sure what would happen with the default `down` method, but deleting manually created permissions with this `down` would be a concern. I'll make that change.

**Pull request merged**